### ISSUE TITLE
Add URL for checking if we can 'lock' Model?

### ISSUE THREAD

**Issue by Enether** [two-man-army/deadline#23]
Models like Course, Lesson, HWTask require locking to be eligible for publishing.
A course depends on all its Lessons and HWTasks to be locked. If a user tries to lock a course but some of its lessons/hwtasks are not locked, he should be able to view which ones they are and it should point them out to him in some sort of error message.
@nemalp do you agree?

Another thing, what do you think about having the ability to concretely LOCK a Course, meaning all Lessons and HWTasks that were not locked would get locked because the Course was locked?
I think it's a somewhat useful feature to have, as it can alleviate going through every lesson/hwtask and locking it separately, although this might cause mistakes from users who accidentally lock it or missed something they wanted to edit in a certain lesson/hwtask.
@nemalp thoughts? (you can take your time)

**Comment by nemalp** [comment]
> Models like Course, Lesson, HWTask require locking to be eligible for publishing...

I agree. We should inform the instructor for those lessons, homework,  etc., which are not marked as completed (locked). 

> Another thing, what do you think about having the ability to concretely LOCK a Course ...

That would be useful, however the course might end up with incomplete lessons etc. Hence, we should warn the instructor to check and confirm if all available course materials are ready to be published before locking the course. 

Either way, the instructor should be able to edit the course at any time. For instance to fix typos etc.

That said, the first option might be unnecessary.

**Comment by Enether** [comment]
Basic editing like descriptions are always allowed. What locking prevents you to do is Adding/Removing Lessons, Adding/Removing Homeworks and other things that could potentially disrupt a student's experience.
i.e imagine a Student just completed his homework and scored 100% but then the Teacher added another HomeworkTask which would make his homework incomplete
Or adding a Lesson after some students have passed the course. You can imagine other scenarios

**Comment by nemalp** [comment]
Probably the best approach would be to force the Teacher to lock the lesson/homework before creating a new one. It doesn't make sense to allow him to start say new lesson without being ready with the previous. If we allow this, it would be a mess. 

What do you think?

**Comment by Enether** [comment]
@nemalp I think that's an awful idea, as we're forcing slow progression.
Most teachers would likely want to create some sort of a draft of all the lessons, to figure out how the actual flow of the course should be. It's hard to build a whole course lesson by lesson without the ability to edit previous ones.
